homeassistant/components/homematicip_cloud/device.py

```diff
@@ -32,6 +33,25 @@ def __init__(self, home, device, post=None):
         self.post = post
         _LOGGER.info("Setting up %s (%s)", self.name, self._device.modelType)
 
+    @property
+    def device_info(self):
+        """Return device specific attributes."""
+        from homematicip.aio.device import AsyncDevice
+        # Only physical devices should be HA devices.
+        if isinstance(self._device, AsyncDevice):
+            return {
+                'identifiers': {
+                    # Serial numbers of Homematic IP device
+                    (homematicip_cloud.DOMAIN, self._device.id)
+                },
+                'name': self._device.label,
+                'manufacturer': self._device.oem,
+                'model': self._device.modelType,
+                'sw_version': self._device.firmwareVersion,
+                'via_hub': (homematicip_cloud.DOMAIN, self._device.homeId),
```

Comment:
Where is this device created?

Reply:
The device referenced in via_hub is created by the sensor of the access point.
With the first start after an update of HA not all devices get a via_hub. 
With secound start of HA all devices have a via_hub, because then the access point already exists as a device and can be linked.
This solution is quite dirty but requires no extra code, othewise the hub has to be created as first device to have all via_hub entries with the first restart.

Reply:
You should define the hub inside the `async_setup_entry`, just like Hue does:

https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/hue/__init__.py#L113-L126

Reply:
hub/ap device creation moved to __init__.py